contracts/Tokenomics.sol

```diff
@@ -1376,12 +1414,19 @@ contract Tokenomics is TokenomicsConstants {
                 totalIncentives = mapUnitIncentives[unitTypes[i]][unitIds[i]].pendingRelativeTopUp;
                 if (totalIncentives > 0) {
                     // Summation of all the unit top-ups and total amount of top-ups per epoch
+                    // TODO: update the formula
                     // topUp = (pendingRelativeTopUp * totalTopUpsOLAS * topUpUnitFraction) / (100 * sumUnitTopUpsOLAS)
-                    totalIncentives *= mapEpochTokenomics[lastEpoch].epochPoint.totalTopUpsOLAS;
-                    totalIncentives *= mapEpochTokenomics[lastEpoch].unitPoints[unitTypes[i]].topUpUnitFraction;
-                    uint256 sumUnitIncentives = uint256(mapEpochTokenomics[lastEpoch].unitPoints[unitTypes[i]].sumUnitTopUpsOLAS) * 100;
                     // Accumulate to the final top-up for the last epoch
-                    topUp += totalIncentives / sumUnitIncentives;
+                    if (mapDonationPoints[lastEpoch].moreDonationPower) {
```

Comment:
This corresponds to the same top-ups calculations as in the write function above.